sdk/search/Azure.Search.Documents/tests/Samples/FieldBuilder/FieldBuilder.cs

```diff
@@ -0,0 +1,451 @@
+        /// the properties of the type supplied.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type for which fields will be created, based on its properties.
+        /// </typeparam>
+        /// <returns>A collection of fields.</returns>
+        public static IList<SearchField> BuildForType<T>() => BuildForType(typeof(T));
+
+        /// <summary>
+        /// Creates a collection of <see cref="SearchField"/> objects corresponding to
+        /// the properties of the type supplied.
+        /// </summary>
+        /// <param name="modelType">
+        /// The type for which fields will be created, based on its properties.
+        /// </param>
+        /// <returns>A collection of fields.</returns>
+        public static IList<SearchField> BuildForType(Type modelType)
+        {
+            bool useCamelCase = SerializePropertyNamesAsCamelCaseAttribute.IsDefinedOnType(modelType);
+            JsonNamingPolicy namingPolicy = useCamelCase
+                ? CamelCaseResolver
+                : DefaultResolver;
+            return BuildForType(modelType, namingPolicy);
+        }
+
+        /// <summary>
+        /// Creates a collection of <see cref="SearchField"/> objects corresponding to
+        /// the properties of the type supplied.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The type for which fields will be created, based on its properties.
+        /// </typeparam>
+        /// <param name="namingPolicy">
+        /// <see cref="JsonNamingPolicy"/> to use.
+        /// This ensures that the field names are generated in a way that is
+        /// consistent with the way the model will be serialized.
+        /// </param>
+        /// <returns>A collection of fields.</returns>
+        public static IList<SearchField> BuildForType<T>(JsonNamingPolicy namingPolicy) => BuildForType(typeof(T), namingPolicy);
+
+        /// <summary>
+        /// Creates a collection of <see cref="SearchField"/> objects corresponding to
+        /// the properties of the type supplied.
+        /// </summary>
+        /// <param name="modelType">
+        /// The type for which fields will be created, based on its properties.
+        /// </param>
+        /// <param name="namingPolicy">
+        /// <see cref="JsonNamingPolicy"/> to use.
+        /// Contract resolver that the SearchIndexClient will use.
+        /// This ensures that the field names are generated in a way that is
+        /// consistent with the way the model will be serialized.
+        /// </param>
+        /// <returns>A collection of fields.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="modelType"/> or <paramref name="namingPolicy"/> is null.</exception>
+        public static IList<SearchField> BuildForType(Type modelType, JsonNamingPolicy namingPolicy)
+        {
+            if (modelType is null)
+            { throw new ArgumentNullException(nameof(modelType));
```

Comment:
Nit - weird whitespace that looks like something auto-formatted it?

Reply:
Also, why not use `Argument.` for these?

Reply:
Because this was meant to be a straight port of track 1. Track 2 `FieldBuilder` likely will, if only in public APIs like this.